assets/src/edit-story/app/story/actions/useSavePost.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../api';
+
+/**
+ * Custom hook to save story.
+ *
+ * @param {boolean}   isSaving Boolean if post is currently saving.
+ * @param {number}    storyId Story post id.
+ * @param {string}    title Story title.
+ * @param {string}    postStatus Post status, draft or published.
+ * @param {Array}     pages Array of all pages.
+ * @param {number}    postAuthor User ID of story author.
+ * @param {string}    slug   The slug of the story.
+ * @param {Function}  setLink Set link in state.
+ * @param {Function}  setPostStatus Set Post Status in state.
+ * @param {Function}  setIsSaving Set saving state.
+ * @return {Function} Function that can be called to save a story.
+ */
+function useSavePost( {
+	isSaving,
+	storyId,
+	title,
+	postStatus,
+	postAuthor,
+	slug,
+	pages,
+	setLink,
+	setPostStatus,
+	setIsSaving,
+} ) {
+	const status = ( postStatus !== 'publish' ) ? 'publish' : postStatus;
+	const { actions: { saveStoryById } } = useAPI();
+
+	/**
+	 * Refresh page to edit url.
+	 *
+	 * @param {number}postId Current story id.
+	 */
+	const refreshPostEditURL = ( postId ) => {
+		const getPostEditURL = addQueryArgs( 'post.php', { post: postId, action: 'edit' } );
+		window.history.replaceState(
+			{ id: postId },
+			'Post ' + postId,
+			getPostEditURL,
+		);
+	};
+
+	const savePost = useCallback( () => {
+		if ( ! isSaving ) {
+			setIsSaving( true );
+			saveStoryById( storyId, title, status, pages, postAuthor, slug ).then( ( post ) => {
+				const { status: thisPostStatus, link } = post;
+				setPostStatus( thisPostStatus );
+				setLink( link );
+				refreshPostEditURL( storyId );
+			} ).catch( () => {
+				// TODO Display error message to user as save as failed.
```

Comment:
Sounds like something we first need UX for. Don't see anything in Figma about displaying such notifications.

Reply:
It makes sense to show an error here. A lot of that work has been done yet for the helpers. If no error message is to be shown, I will remove this TODO later.

Reply:
What I meant is UX for _how_ it should be displayed. It makes sense to have error messages for sure.